gtsam/hybrid/HybridBayesNet.h

```diff
@@ -101,6 +87,35 @@ class GTSAM_EXPORT HybridBayesNet : public BayesNet<HybridConditional> {
         std::make_shared<HybridConditional>(std::move(conditional)));
   }
 
+  /**
+   * @brief Add a conditional to the Bayes net.
+   * Implicitly convert to a HybridConditional.
+   *
+   * E.g.
+   * hbn.push_back(std::make_shared<DiscreteConditional>(m, "1/1"));
+   *
+   * @tparam CONDITIONAL Type of conditional. This is shared_ptr version.
+   * @param conditional The conditional as a shared pointer.
+   */
+  template <class CONDITIONAL>
+  void push_back(std::shared_ptr<CONDITIONAL> conditional) {
```

Comment:
This is good. Although check what we do in GaussianBayesNet. I think we use const& to a shared pointer there. My comment before about const& was about a ref to the underlying object, not the shared pointer.

Reply:
`GaussianBayesNet` inherits from `FactorGraph` which has this method (which I followed), so no const& for the shared pointer:
```cpp
/// Add a factor directly using a shared_ptr.
template <class DERIVEDFACTOR>
IsDerived<DERIVEDFACTOR> push_back(std::shared_ptr<DERIVEDFACTOR> factor) {
  factors_.push_back(std::shared_ptr<FACTOR>(factor));
}
```

I should add const& since it provides a small performance benefit.